rfp-proposals/rfp-proposal-reputation-system.md

```diff
@@ -0,0 +1,326 @@
+* Tutorial workflow with demo usage.
+* Well-documented, human-readable codebase.
+
+
+### Specifics of service:
+
+
+#### /get
+The users will be able to query the following for a particular miner: 
+* Metrics mentioned in RFP: 
+    * Number and % (where applicable) of deals with no penalties
+    * Number and % (where applicable) of slashed contracts
+    * Number and % (where applicable) of dropped contracts
+    * Number and % of failed deals (if possible) -- i.e. the number of deals that were attempted but failed, possibly due to this miner having spotty connectivity
+    * % of retrieval requests successfully fulfilled
+    * Number of files retrieved
+    * Lifetime data stored
+    * Lifetime data retrieved
+* Metrics that we have thought of: 
+    * Collateral: 
+        * Reason: Over collateralization may signal more skin in the game from the miner. 
+    * Storage Price (in FIL/byte/block): 
+        * % deviation of storage price from: 
+            * mean of all storage prices on the network. 
+            * median of all the storage prices of the network. 
+        * Reason: Some miners may be overpricing/underpricing their storage with respect to the market. 
+    * % of miner storage being used.  
+        * Formula = used storage/total storage. 
+        * Reason: Miners consistently hovering around > 80% may signal more trustworthiness. 
+    * Uptime:
+        * Reason: This is a more generic metric to track then failed deals. A miner may be up consistently but not getting any deals due to other reasons. 
+    * Miner Proactiveness: 
+        * Delta of time since Filecoin published an upgrade for node software and the miner updated itself. 
+        * Reason: 
+            * Smaller the delta, the more proactive the miner. 
+            * Miners who don't update are risky. 
+
+
+   
+### Specifics of web app: 
+
+
+#### Home page:
+Displaying  list of miners currently providing space for allocation with the following info:
+* Miner
+* Miner Score
+* Used Storage
+* Available storage
+* Storage Price
+* Valid till
+* Status
+
+ 
+![](https://i.imgur.com/OsVK6v2.png)
+
+^ Early Prototype. 
+
+<br><br>
+
+#### Miner Individual page:
+* Miner stats card containing information:
+    * Miner Address
+    * Filecoin version
+    * Collateral 
+    * Miner geographic location
+    * Storage Metrics: Used/Available/Total Storage. 
+
+* Miner Score Table: (10 days average):
+    * [Final/Average Score](#How-will-scores-be-calculated?)
+    * Individual characteristics score:
+        * Latency
+        * % of successful deals
+        *  %  of slashed and dropped contracts
+        * etc
+
+* Comparison with storage miners with the same score:
+    * Storage price
+    * Collateral
+    * Est. uptime
+    * Stored data
+
+* Substantiated feedback:
+    * identifying the user:
+        * a message-signing-based authentication mechanism with a user's public address as their identifier.
```

Comment:
How will the user do this? One of you docs mentions Metamask but that doesn't exist for Filecoin.

Reply:
Yup, we also came to this conclusion and hence we were thinking of going with keybase until the time the community develops Filecoin Wallet. 

Thought process: 
If we do get funded, we see an overlap in terms of timelines that when get to the development of user auth module, we would likely be able to leverage the work being done by the team(s) that are parrellely developing wallets for Filecoin and work with them to integrate it in our service in time for mainnet launch.